metadata-ingestion/src/datahub/ingestion/api/source_helpers.py

```diff
@@ -173,6 +175,40 @@ def auto_materialize_referenced_tags(
         ).as_workunit()
 
 
+def auto_lowercase_urns(
+    stream: Iterable[MetadataWorkUnit], enabled: bool = False
+) -> Iterable[MetadataWorkUnit]:
+    """Lowercase all dataset urns"""
+
+    if not enabled:
+        return stream
+
+    for wu in stream:
+        try:
+            urn = Urn.create_from_string(wu.get_urn())
+            if urn.get_type() == DatasetUrn.ENTITY_TYPE:
+                dataset_urn = DatasetUrn.create_from_string(str(urn))
+                lowercased_urn = DatasetUrn.create_from_ids(
```

Comment:
this only fixes the urn, but won't fix lineage edges or things

Let's use the `lowercase_dataset_urns` helper method instead https://github.com/datahub-project/datahub/blob/cbd8e14b78f720a3584b0390d272a8a4321109de/metadata-ingestion/src/datahub/utilities/urns/urn_iter.py#L123